packages/netlify/src/ssr-function.ts

```diff
@@ -24,7 +25,13 @@ export const createExports = (manifest: SSRManifest, { middlewareSecret }: Args)
 		cacheOnDemandPages: boolean;
 		notFoundContent?: string;
 	}) {
+		// The entrypoint is created in `.netlify/v1/build` so we need to go up two levels to get to the root
+		const root = fileURLToPath(new URL('../../', import.meta.url));
 		return async function handler(request: Request, context: Context) {
+			// This entrypoint will be deep inside the directory structure below cwd
+			// We chdir to the root so that we can resolve the correct paths at runtime
+			process.chdir(root);
```

Comment:
Are we sure it can't break any existing app?

Reply:
As far as I can work out, existing apps that rely on cwd would either have unchanged behaviour (if they're not a monorepo), or would already be broken (if they are using a monorepo)

Reply:
If a site uses filesystem functions at runtime, this is the only way it can correctly load the files

Reply:
I don't think using filesystem functions is something Astro supports in the runtime? Files could be moved during bundling and it wouldn't work later.

Reply:
We don't prevent people using fs functions, so I'd imagine lots of people do. This PR was prompted by an issue where a library was trying to load i18n data and it wasn't found. This part fixes precisely that issue: of files being moved during bundling. Without that there's not a lot of point to allowing users to include files in the bundle, because the user code won't be able to find them.

Reply:
I don't feel like fs functions are a common thing especially when dealing in a bundled runtime. There's Vite APIs like `import.meta.glob`, `?url`, and `?raw` for them. Fs functions are also likely to fail elsewhere than only on Netlify, and so we shouldn't support and encourage them.

I figured the benefit with `includeFiles` and `excludeFiles` are like Vercel's option, because nft has trouble sometimes detecting every file.

I'm not really comfortable with keeping this in the adapter. It feels surprising to me that it could change the cwd internally and on every request.

Reply:
They _are_ like Vercel's option. In fact they use almost the same implementation. With libraries that use fs functions, they usually accept a directory as an option or resolve relative to cwd. It's very common for i18n libraries such as i18next, for example.

Reply:
If we're about to do a major I can do a breaking changeset for the chdir part.